src/connections.jl

```diff
@@ -50,15 +50,26 @@ mutable struct Connection
     func_map::PQConversions
 
     "True if the connection is closed and the PGconn object has been cleaned up"
-    closed::Bool
+    closed::Atomic{Bool}
+
+    "Lock for thread-safety"
+    lock::Mutex
```

Comment:
Why `Mutex` and not `SpinLock`?

Reply:
Basically I saw: 
> See also Mutex for a more efficient version on one core or if the lock may be held for a considerable length of time.
and those conditions might hold here? Not sure, is there a reason a spin lock would be better?

Reply:
Also psycopg2 uses a mutex: https://github.com/psycopg/psycopg2/blob/master/psycopg/connection.h#L87

but maybe they have additional constraints, like compatibility.

Reply:
I was not aware the Mutex had those advantages.
Seems legit.
I'm used to `SpinLock` being better because it better under contention (apparently)
but I guess this is low contention?

If not holding it for long, then locking and unlocking a mutex takes almost twice as long as for a `SpinLock`.
(I just benchmarked)

Reply:
I don't care much about the lock timing if it's less than a millisecond. Network I/O will always dominate.

If we had to `lock` for every `PQgetvalue` call I would definitely care.